lib/queue.go

```diff
@@ -163,12 +170,31 @@ func (q *RequestQueue) Queue(req *http.Request, res *http.ResponseWriter, path s
 		"method": req.Method,
 	}).Trace("Inbound request")
 
+	var abort int
+	abortHeader := req.Header.Get("X-RateLimit-Abort-After")
+	if abortHeader != "" {
+		valParsed, err := strconv.ParseInt(abortHeader, 10, 64)
+		if err != nil {
+			return err
+		}
+
+		abort = int(valParsed)
+	} else {
+		abort = EnvGetInt("RATELIMIT_ABORT_AFTER", -1)
```

Comment:
For consistency and due to this being a hot path, can we move the EnvGetInt call to main.go and pass it from queue_manager to queue in the same way that bufferSize is passed and store it in the Queue struct?

Reply:
I was considering this, but saw the following code using it directly:

https://github.com/germanoeich/nirn-proxy/blob/6cfb09a8f5a86674034d48768945c5c08e96518d/lib/queue.go#L365

This is definitely a *hotter* codepath, but I suppose you may want to consider changing that one too in another PR/commit.